beacon-chain/rpc/statefetcher/state_fetcher.go

```diff
@@ -0,0 +1,108 @@
+package statefetcher
+
```

Comment:
Optional:

Can name this file `fetcher.go` instead. The folder name is `statefetcher` which already implies `state`